include/swift/AST/Attr.h

```diff
@@ -181,6 +181,13 @@ enum class DeclKind : uint8_t;
 
   /// Represents one declaration attribute.
 class DeclAttribute : public AttributeBase {
+  // XXX -- We cannot use DeclKind due to layering.
```

Comment:
I don't think we use "XXX" anywhere else in the code base. Please use "Note:" or "FIXME:" instead, depending on whether you think this should ever get fixed.

As for the actual comment, I think you should point out that it's okay if this isn't 1-to-1 with DeclKind, since it's just used to build up DeclAttrOptions. It should also probably be an `enum class` to not infect the DeclAttribute namespace.

Reply:
That's interesting, because I had always thought I was the first one to use "XXX" or "xxx" in that fashion. It was a hack, and I agree with @jrose-apple about not introducing it into this code base. But I would love to be able to trace down the meme. Did I copy it from somewhere and just forget? Did a chain of others copy it from me??? I know I used it in the Self VM back in the late 80's.

Reply:
It's a perfectly normal convention these days, and I've seen it in many codebases. I just don't want to add more such conventions to *this* codebase.

Reply:
HACK is around, but this isn't a hack. If it were, you should just move the DeclKind enum into its own header.

Reply:
I've elaborated the comment and dropped the use of a comment tag entirely.